### PR TITLE
Refactor endpoints

### PR DESCRIPTION
**Pull request by bernardolk**
- Changed API from "match_spectra_from_path" to "match_spectra" and now accepting 3 types of MGF sources (path to file, List[Spectrum] and StringIO for MGF files opened with another methodology)
- Refactored the Endpoint class to hold all logic for matching spectra (it is called now SpectrumMatching) and trimmed down Spec2Vec and MS2DeepScore into shallow classes with only endpoint string differing from each other
- Renamed other methods and refactored tests
- Included "precursor_mz" as a valid metadata key

### CODE REVIEW

File: omigami/__init__.py [hunk 1]
```diff
@@ -12,8 +12,7 @@
 
 __version__ = get_versions()["version"]
 del get_versions
-from omigami.spec2vec import Spec2Vec
-from omigami.ms2deepscore import MS2DeepScore
+from .spectra_matching import MS2DeepScore, Spec2Vec
 from omigami.plotting import MoleculePlotter
```
Comment (cereniyim) [inline omigami/__init__.py:16]: pycharm says that this import is not used, should we remove it?

File: omigami/spectra_matching.py [hunk 2]
```diff
@@ -26,27 +36,91 @@
...
-        mgf_path: str,
+        source: Union[str, list[Spectrum], StringIO],
         n_best: int,
         include_metadata: List[str] = None,
         ion_mode: str = "positive",
-    ) -> List[pd.DataFrame]:
+    ):
         """
-        Finds the N best matches for spectra in a local mgf file
+        Finds the N best matches for spectra in a local mgf file using spec2vec algorithm.
```
Comment (cereniyim) [inline omigami/spectra_matching.py]: don't we use this class for `ms2deepscore` as well?

Comment (cereniyim) [reply omigami/spectra_matching.py]: yep we do, so I suggest adding it to docstring

**Review by cereniyim — COMMENTED** [review]
I added some comments

File: omigami/__init__.py (hunk 1 above)
Comment (bernardolk) [reply omigami/__init__.py:16]: No, because we want `from omigami import MS2DeepScore` (e.g.) to work

**Pull request merged**